swarm/storage/chunker.go

```diff
@@ -483,8 +483,12 @@ func (s *LazyChunkReader) Seek(offset int64, whence int) (int64, error) {
 	case 1:
 		offset += s.off
 	case 2:
-		if s.chunk == nil {
-			return 0, fmt.Errorf("seek from the end requires rootchunk for size. call Size first")
+		if s.chunk == nil { //seek from the end requires rootchunk for size. call Size first
+			quitC := make(chan bool)
+			_, err := s.Size(quitC)
+			if err != nil {
+				return 0, fmt.Errorf("seek from the end requires size. call to Size() failed: %v", err.Error())
```

Comment:
Btw, `nil` is a valid channel value. You can use it instead of passing a channel that is never written to.